include/Factory.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "King.h"
+#include "Queen.h"
+#include "Rook.h"
+#include "Bishop.h"
+#include "Knight.h"
+#include "Pawn.h"
+#include "macros.h"
+
+#include <functional>
+#include <memory>
+#include <map>
+
+
+class Factory {
+
+public:
+    std::unique_ptr<Piece> buildQueen(bool isWhite) {
+        return std::make_unique<Queen>(isWhite);
+    }
+
+    std::unique_ptr<Piece> buildKing(bool isWhite) {
+        return std::make_unique<King>(isWhite);
+    }
+
+    std::unique_ptr<Piece> buildRook(bool isWhite) {
+        return std::make_unique<Rook>(isWhite);
+    }
+
+    std::unique_ptr<Piece> buildKnight(bool isWhite) {
+        return std::make_unique<Knight>(isWhite);
+    }

```

Comment:
It's a good practice to separate the implementation of class member functions into a separate source file (.cpp) rather than including them in the header file. This way, you can keep the header file clean and only include necessary declarations.